src/sentry/middleware/staff.py

```diff
@@ -0,0 +1,57 @@
+from django.conf import settings
+from django.utils.deprecation import MiddlewareMixin
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry.auth.staff import Staff, logger
+
+
+class StaffMiddleware(MiddlewareMixin):
+    def process_request(self, request: Request):
+        # This avoids touching user session, which means we avoid
+        # setting `Vary: Cookie` as a response header which will
+        # break HTTP caching entirely.
+        self.__skip_caching = request.path_info.startswith(settings.ANONYMOUS_STATIC_PREFIXES)
+
+        if self.__skip_caching:
+            return
+
+        stf = Staff(request)
+
+        request.staff = stf
+        request.is_staff = lambda: request.staff.is_active
```

Comment:
this is present in `SuperuserMiddleware` to support some legacy usage, but since this is a new middleware i don't think we need it

most places check for `request.user.is_superuser`

Reply:
oh right, I should remove this